react/getting_started_with_react/rendering_techniques.md

```diff
@@ -0,0 +1,299 @@
+
+We have also created a different component for the `<li>` element called `<ListItem />`, which also accepts a `props`, and using `props.animal` to render the text. This kind of pattern is very useful because a component can have their own state, unlike a regular element. It should now render the same thing. 
+
+### Conditionally Rendering UI
+
+Let us make some decisions within our component. What if we only want to render an animal that starts with the letter L? To make these decisions, we would use some sort of conditional expression. Let us continue using the code above, but for brevity's sake, we will be removing the `<ListItem />` component.
+
+#### Using Ternary Operator
+
+One way to conditionally render an element is with a ternary operator, using a boolean value to decide what to render:
+
+~~~javascript
+function List(props) {
+  return (
+    <ul>
+      {props.animals.map((animal) => {
+        return animal.startsWith("L") ? <li key={animal}>{animal}</li> : null;
+      })}
+    </ul>
+  );
+}
+
+function App() {
+  const animals = ["Lion", "Cow", "Snake", "Lizard"];
+
+  return (
+    <div>
+      <h1>Animals: </h1>
+      <List animals={animals} />
+    </div>
+  );
+}
+~~~
+
+We are using the String method `startsWith` to check if the `animal` starts with the letter L. This method either returns true or false.
+
+If the animal starts with the letter L, then we return the `<li>` element, which renders the particular animal. Otherwise, we return `null` to indicate that no element will be rendered.
+
+#### Using the && Operator
+
+Another quick way of conditionally rendering an element is by using the && operator.
+
+~~~javascript
+function List(props) {
+  return (
+    <ul>
+      {props.animals.map((animal) => {
+        return animal.startsWith("L") && <li key={animal}>{animal}</li>;
+      })}
+    </ul>
+  );
+}
+
+function App() {
+  const animals = ["Lion", "Cow", "Snake", "Lizard"];
+
+  return (
+    <div>
+      <h1>Animals: </h1>
+      <List animals={animals} />
+    </div>
+  );
+}
+~~~
+
+We will leverage the return value of `startsWith` With the && operator. If the result of the `startsWith` function is `true`, then it returns the second operand, which is the `<li>` element and renders it. Otherwise, if the condition is `false` it just gets ignored.
+
+In JSX, values like `null`, `undefined`, and `false` do not render anything, and you might ask aren't they falsy values? So you might think a value like `0` or an empty string does the same thing. It is a common pitfall. They are valid in JSX and will be rendered completely fine, so be sure to be aware of that!
+
+We are not limited to rendering an element or nothing with `null` we can also decide whether to render an element or another element.
```

Comment:
I think we can remove this sentence altogether. The next section already has an example that showcases this.

Reply:
It seems more appropriate to be in the next section instead of where it's currently at. I think it's a nice small description on what we're going to do next.

Or do you think this already does that subtly?
> We will frequently be dealing with lists in the future, and we also need to consider what to render if the list is empty or does not exist at all. You certainly would not want to see a blank page, would you? Let us try to implement that:

Would be glad to know more of your opinion, it's also fine to remove it.

Reply:
I feel the code examples are sufficient to show that we don't have to render null but can use other elements/components. As such, removing the sentence gets my vote.

Reply:
Gotcha. Done!